src/task-actions.ts

```diff
@@ -173,6 +174,9 @@ export async function starknetCompileAction(args: TaskArguments, hre: HardhatRun
                 disableHintValidation: args.disableHintValidation
             });
 
+            if (hre.userConfig?.starknet?.recompile) {
```

Comment:
Shouldn't the `if` be removed? Like if the user compiles, then turns on recompilation, then runs tests, the recompilation would be done unnecessarily. I.e. we always want to store the cache after compilation.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/blob/3c16ef595ce68cf479a1a72f3bee82e182226a25/src/task-actions.ts#L175

Now, we update cache after every recompilation